lib/hooks/useSearchRepos.ts

```diff
@@ -0,0 +1,27 @@
+import useSWR, { Fetcher } from "swr";
```

Comment:
Leveragin useSWR so that repository queries can be cached. It would be good to leverage this for repository search elsewhere in the app, e.g. insights pages.